### PR TITLE
Prepare backwards compatible usage of optional C++ toolchain

### PR DESCRIPTION
**Pull request by fmeum**
By adding an alias and a no-op keyword argument now, rulesets can start using the new optional form of `find_cpp_toolchain` now and benefit from the toolchain actually being optional once that has been implemented in a future version of Bazel.

Work towards #16966

### CODE REVIEW

**Comment by fmeum** [comment]
@oquenchil @katre

**Review by katre — APPROVED** [review]

**Comment by katre** [comment]
I'll let @oquenchil review and approve before merging, but this looks good.

**Comment by fmeum** [comment]
@bazel-io flag

**Review by oquenchil — APPROVED** [review]

**Comment by kshyanashree** [comment]
@bazel-io fork 6.1.0

**Pull request closed without merging**
